FirebaseAuth/Sources/Swift/Backend/RPC/CreateAuthURIResponse.swift

```diff
@@ -45,6 +45,6 @@ class CreateAuthURIResponse: AuthRPCResponse {
     registered = dictionary["registered"] as? Bool ?? false
     forExistingProvider = dictionary["forExistingProvider"] as? Bool ?? false
     allProviders = dictionary["allProviders"] as? [String]
-    signinMethods = dictionary["signinMethods"] as? [String]
+    signinMethods = dictionary["signinMethods"] as? [String] ?? []
```

Comment:
It's technically an API change to return an empty array instead of nil, so worth calling out in the release notes.

Reply:
I think this PR may actually be re-aligning the behavior with the API contract:
https://github.com/firebase/firebase-ios-sdk/blob/486a0a03c197a267a4a54d5f0077e1703b197588/FirebaseAuth/Sources/Swift/Auth/Auth.swift#L279-L281

In the current behavior, `(nil, nil)` was being passed to the completion handler which caused the crash.

Reply:
I will merge this now, but can address anything else post-merge.

Reply:
Won't those using the callback version of the API now get `[], nil` instead of `nil,nil`

Reply:
Yes, `[], nil`. I'm thinking that this is a valid combo as opposed to `nil, nil`.

Reply:
Also, I just manually tested in a project with email enumeration protection turned **off**, and `nil, nil` gets returned there and crashes.

Reply:
Agreed, but it's possible there may be apps checking for `nil,nil` since it doesn't crash when using the callback api

Reply:
True, ok I'll point this out in release notes.

One different approach to this PR is to preserve `nil, nil` as a valid combo and change the async wrapper to return a `[String]?` instead `[String]`. 

I think this PR with the corresponding release note is the better option.

Reply:
Yes. I would suspect that in practice, code would just not exercise the `nil` case and just loop zero times iterating the empty list.

Reply:
So unlikely anything will break if the release note gets missed

Reply:
Done in d855c25